### ISSUE TITLE
Can't open Debugger and React Dev Tool 

### ISSUE THREAD

**Issue by panpan31415** [facebook/react-native#34624]
### Description

Can't open Debugger in iOS simulator, if you try to open you will get 
`Running "AwesomeProject" with {"rootTag":1,"initialProps":{}}
info Opening flipper://null/React?device=React%20Native...
error Browser exited with error:, Error: invalid url, missing http/https protocol
`

### Version

0.70.0

### Output of `npx react-native info`

info Fetching system and libraries information...
System:
    OS: macOS 12.5.1
    CPU: (8) arm64 Apple M1
    Memory: 153.98 MB / 16.00 GB
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 16.17.0 - ~/.nvm/versions/node/v16.17.0/bin/node
    Yarn: 1.22.19 - /opt/homebrew/bin/yarn
    npm: 8.15.0 - ~/.nvm/versions/node/v16.17.0/bin/npm
    Watchman: 2022.09.05.00 - /opt/homebrew/bin/watchman
  Managers:
    CocoaPods: 1.11.3 - /opt/homebrew/bin/pod
  SDKs:
    iOS SDK:
      Platforms: DriverKit 21.4, iOS 15.5, macOS 12.3, tvOS 15.4, watchOS 8.5
    Android SDK: Not Found
  IDEs:
    Android Studio: 2021.2 AI-212.5712.43.2112.8609683
    Xcode: 13.4.1/13F100 - /usr/bin/xcodebuild
  Languages:
    Java: 18.0.2.1 - /usr/bin/javac
  npmPackages:
    @react-native-community/cli: Not Found
    react: 18.1.0 => 18.1.0 
    react-native: 0.70.0 => 0.70.0 
    react-native-macos: Not Found
  npmGlobalPackages:
    *react-native*: Not Found

### Steps to reproduce

before I install the latest version of react native, 
I followed the [guide](https://reactnative.dev/docs/environment-setup#creating-a-new-application) of RN official site
`npm uninstall -g react-native-cli @react-native-community/cli`
then 
`npx react-native init AwesomeProject`
after app build successfully, my simulator pops up and run the app. 
then click iOS simulator -> Device -> shake
click `Open Debugger`.  You will see the flowing message in terminal
`info Opening flipper://null/React?device=React%20Native...
error Browser exited with error:, Error: invalid url, missing http/https protocol
`

### Snack, code example, screenshot, or link to a repository

![image](https://user-images.githubusercontent.com/11925038/189001812-70b5b3e9-0c51-403d-9f14-e261372ac296.png)

**Comment by cuongvoong** [comment]
#34615

**Comment by panpan31415** [comment]
> #34615

**Comment by rohitthakre** [comment]
Browser exited with error:, Error: invalid url, missing http/https protocol

**Comment by prasanna-colan** [comment]
> Browser exited with error:, Error: invalid url, missing http/https protocol

Got any luck?

**Comment by euphratesriver0216** [comment]
In my opinion, flipper has not been updated yet.
Should we just wait for the update?

**Comment by shanhassan06** [comment]
You have to Set  :hermes_enabled => false in podfile

**Comment by shanhassan06** [comment]
You should try
 :hermes_enabled => false 
in podfile

**Comment by Basir-PD** [comment]
1+

**Comment by Basir-PD** [comment]
I don't know if this is the ideal solution, but I ended up temporarily disabling Hermes `:hermes_enabled => false` and then enabling it back after debugging.

**Comment by ahmad6242** [comment]
:hermes_enabled => false 
did not work. 
having same issue,

**Comment by dariabr** [comment]
Any updates?

**Comment by TiagoAndreAlmeida** [comment]
any new ?

**Comment by davidmaogomezz** [comment]
same problem here

**Comment by babar7** [comment]
https://reactnative.dev/docs/hermes#debugging-js-on-hermes-using-google-chromes-devtools
use this to enable hermes-chromes-devtools

**Comment by dariabr** [comment]
Update: 
:hermes_enabled => false
in podfile

then Install pods cd ios && pod install && cd ..
Build App npx react-native run-ios
worked for me

**Comment by anuragomnics** [comment]
> :hermes_enabled => false did not work. having same issue,

you need to re-install pods after any change in podfile

**Comment by camilogore99** [comment]
> hermes_enabled

i love you ❤️

**Comment by thachsteven** [comment]
> cd ios && pod install && cd ..

you saved me

**Comment by lauhon** [comment]
Turning off Hermes resolves the issue, but you might want to keep Hermes enabled because of the related performance benefits.

You can still use Browser debugging, but instead of opening from inside the simulator you have to follow [this guide](https://reactnative.dev/docs/hermes#debugging-js-on-hermes-using-google-chromes-devtools)

**Comment by AugustoAleGon** [comment]
@lauhon Do you know how to use `React Native Debugger`? I am having a hard time actually make it work on that tools.

**Comment by lauhon** [comment]
> @lauhon Do you know how to use `React Native Debugger`? I am having a hard time actually make it work on that tools.

Yes, but I think this thread is not the place to discuss how the Debugger works in general. Have a look at the Docs on https://reactnative.dev/ - there is everything you need basically :)

**Comment by Rafazor** [comment]
> 

This one works, but it's super bad that we need to do magic fixes to see a console log in the debugger :(

**Comment by BulletButcher88** [comment]
Where in the Podfile do we place `:hermes_enabled => false`
I am getting errors if I place it within the file

**Comment by niuyunfei01** [comment]
> 更新：: hermes_enabled => podfile 中的 false
> 
> 然后安装 pods cd ios && pod install && cd .. Build App npx react-native run-ios 为我工作

very good

**Comment by ivantrave** [comment]
Setting :hermes_enabled => false did not work for me.
After I changed that I executed:
`cd /ios `
`pod install` (even tried `pod deintegrate` before this)
`cd ..`
`npm start`

Same error: `error Browser exited with error:, Error: invalid url, missing http/https protocol`

I am using "react-native": "0.71.3",

Any suggestion?

**Comment by lechinoix** [comment]
When following the docs for hermes, I only have the logs of the console in the debugger. I definitely need those Flipper or React Native debugger features (network inspect, JS breakpoints, React Navigation debugger). This makes it really hard for me to work on iOS locally because I need to disable Hermes, pod install and when finished to enable Hermes pod install. This is a very long process and it makes me work without debugger most of the time so real poor Dev experience.

I think this should be fixed for real, pretty surprised that this issue is closed.

**Comment by shweta-amaha** [comment]
i am not using flipper, but using hermes instead, 
on doing steps from react native documen,t hermes devtool connection from "chrome://inspect/#devices" and addinglocalhost:8081 , i was able to see logs on devtools but, 
whenever hot or live reloading happens, or i kill the app and load it again, it disconnects from the metro,
any solution ?

**Comment by sameersitre** [comment]
> Update: :hermes_enabled => false in podfile
> 
> then Install pods cd ios && pod install && cd .. Build App npx react-native run-ios worked for me

Thanks! its working.

**Comment by Rache-lBronfman** [comment]
When I set hermes_enabled to false, the debugger worked for me, 
but when I'm trying to make API request (using axios) the app and the  debugger stop working

**Comment by ghorbani-m** [comment]
In my case, I just remove the app from the simulator and rebuild the app again. (Also I ran pod install before build but I don't know it have affect or not)

**Comment by R4DIC4L** [comment]
In my case, I cannot make the debugger or Network tab work in any way in RN 0.70.10 for iOS. And it doesn't matter if I have hermes_enabled on true or false (I have flipper disabled because of react-native-firebase for push notifications). And when you need to debug an issue reproducing only in iOS, this is really frustrating, as I currently see no option for this.

When trying to use the Open debugger option (http://localhost:8081/debugger-ui/) I still get the said error (but I don't understand why is flipper requested for debugging when flipper is disabled ??):
**info** Opening flipper://null/Hermesdebuggerrn?device=React%20Native...
**error** Browser exited with error:, Error: invalid url, missing http/https protocol
Why is this issue altogether closed? Is the non-Flipper debugger working again in the newer versions with hermes enabled? 

When using chrome://inspect following the guide with hermes enabled I see and can press inspect for Hermes React Native, but only see the console logging, no Sources are available and no Network tab.

I usually run my app from Xcode.
Now trying to disable hermes as well and run command npx react-native run-ios, it gives a BUILD FAILED for me, so cannot use this approach at all.  I run on MacOS Ventura 13.3.1 with Xcode Version 14.3.
**error** Failed to build iOS project. We ran "xcodebuild" command but it exited with error code 65. To debug build logs further, consider building your app with Xcode.app, by opening <AppName>.xcworkspace.

When trying to use without hermes from Xcode I can get the DevTools open from the Open in Chrome option, but I still don't see the code, even after reloading several times.
![image](https://github.com/facebook/react-native/assets/7274664/bf393a91-aac3-4e35-b201-495f1ef01f89)

Any other options available for debugging react native apps in iOS? Because I can see no debugging option works for me when Flipper is disabled.

**Comment by Chenjiujiu** [comment]
Yes, I had the same problem, and now, I use react-native firebase, which means I don't have any debugging tools available

**Comment by Chenjiujiu** [comment]
> 就我而言，我无法使调试器或网络选项卡在适用于iOS的RN 0.70.10中以任何方式工作。而且我是否hermes_enabled真或假都没有关系（由于推送通知的 react-native-firebase 我禁用了鳍状肢）。当您需要调试仅在iOS中重现的问题时，这确实令人沮丧，因为我目前看不到任何选择。
> 
> 尝试使用打开调试器选项（ [http://localhost:8081/debugger-ui/ ](http://localhost:8081/debugger-ui/)）时，我仍然收到上述错误（但我不明白为什么在禁用鳍状肢时请求鳍状肢进行调试？？ **信息打开 ** flipper://null/Hermesdebuggerrn?device=React%20Native... **error**错误 浏览器退出并显示错误：，错误：网址无效，缺少 http/https 协议 为什么这个问题完全关闭了？非 Flipper 调试器是否在启用了爱马仕的较新版本中再次工作？
> 
> 在启用 Hermes 的情况下按照指南使用 chrome://inspect 时，我看到并且可以按 Hermes React Native 的检查，但只能看到控制台日志记录，没有可用的源，也没有网络选项卡。
> 
> 我通常从 Xcode 运行我的应用程序。 现在尝试禁用爱马仕并运行命令 npx 反应本机运行-ios，它为我提供了一个构建失败，所以根本无法使用这种方法。 我在MacOS Ventura 13.3.1上运行，Xcode版本为14.3。 **错误 ** 无法构建 iOS 项目。我们运行了“xcodebuild”命令，但它以错误代码 65 退出。若要进一步调试生成日志，请考虑通过打开 .xcworkspace 来使用 Xcode.app 生成应用。
> 
> 当尝试在没有 Xcode 的爱马仕的情况下使用时，我可以从“在 Chrome 中打开”选项打开 DevTools，但即使重新加载了几次，我仍然看不到代码。 ![image](https://user-images.githubusercontent.com/7274664/250017117-bf393a91-aac3-4e35-b201-495f1ef01f89.png)
> 
> 还有其他可用于在iOS中调试反应本机应用程序的选项吗？因为当 Flipper 被禁用时，我看不到任何调试选项对我有用。

I ran into the same problem, which resulted in me not having any debugging tools available

**Comment by joeljerushan** [comment]
This problem is is bit tricky we can disable herms but those who enabled herms settings for RNFirebase will ended up with issues, using *Debugging JS on Hermes using Google Chrome's DevTools* is works fine but every-time i start working i have to do the same, is there any permanent solution for this issue available ?

**Comment by corysimmons** [comment]
After trying a million things, I installed Flipper and its logs seemed to work fine (even with Hermes).

<img width="1126" alt="image" src="https://github.com/facebook/react-native/assets/224636/c96cdadf-ec0d-4a0b-8314-08409cfdc968">

**Comment by juliuspopa** [comment]
Installed this [https://github.com/gusgard/react-native-devsettings](https://github.com/gusgard/react-native-devsettings) and it works with Hermes enabled. Breakpoints in dev tools work as well as `debugger` statements in your JS code.

**Comment by Isaacmeedinaa** [comment]
@juliuspopa this works. thanks.
